src/profile-logic/flame-graph.js

```diff
@@ -0,0 +1,77 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+// @flow
+import type { Thread } from '../types/profile';
+import type { Milliseconds } from '../types/units';
+import type {
+  IndexIntoCallNodeTable,
+  CallNodeInfo,
+} from '../types/profile-derived';
+
+import * as CallTree from './call-tree';
+
+export type FlameGraphTiming = Array<{
+  start: Milliseconds[],
+  end: Milliseconds[],
+  callNode: IndexIntoCallNodeTable[],
+  length: number,
+}>;
+
+/**
+ * Build a FlameGraphTiming table from a given thread.
+ *
+ * @param {object} thread - The profile thread.
+ * @param {object} callNodeInfo - from the callNodeInfo selector.
+ * @param {CallTree} callTree - The call tree.
+ * @return {array} flameGraphTiming
+ */
+export function getFlameGraphTiming(
+  thread: Thread,
+  callNodeInfo: CallNodeInfo,
+  callTree: CallTree.CallTree
+): FlameGraphTiming {
+  const timing = [];
+  const timeOffset = [];
+
+  let stack = callTree.getRoots().map(nodeIndex => ({ nodeIndex, depth: 0 }));
+
+  while (stack.length) {
+    const { depth, nodeIndex } = stack.pop();
+    const totalTime = callTree._callNodeTimes.totalTime[nodeIndex];
```

Comment:
`getNode` returns data formatted as strings, so for now I just reached into this table directly but it should of course be dealt with in a better way. Can we extend the object returned from `getNode` to include numeric data, or perhaps split the method into two, one with formatted data and one without?

Reply:
Feel free to modify `getNode` to include the data you need, or wherever feels logical to you.